src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using Microsoft.Azure.SignalR.Management.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal static class DependencyInjectionExtensions //TODO: not ready for public use
+    {
+        /// <summary>
+        /// Adds the essential SignalR Service Manager services to the specified services collection and configures <see cref="ServiceManagerOptions"/> with configuration instance registered in service collection.
+        /// </summary>
+        public static IServiceCollection AddSignalRServiceManager(this IServiceCollection services)
+        {
+            services.AddSingleton<IConfigureOptions<ServiceManagerOptions>, ServiceManagerOptionsSetup>();
+            services.AddSingleton<ServiceManagerOptionsSetup>()
+                .AddSingleton<IConfigureOptions<ServiceManagerOptions>>(sp => sp.GetService<ServiceManagerOptionsSetup>())
+                .AddSingleton<IOptionsChangeTokenSource<ServiceManagerOptions>>(sp => sp.GetService<ServiceManagerOptionsSetup>());
+            return services.AddSignalRServiceManagerCore();
+        }
+
+        /// <summary>
+        /// Adds the essential SignalR Service Manager services to the specified services collection and registers an action used to configure <see cref="ServiceManagerOptions"/>
+        /// </summary>
+        public static IServiceCollection AddSignalRServiceManager(this IServiceCollection services, Action<ServiceManagerOptions> configure)
+        {
+            services.Configure(configure);
+            return services.AddSignalRServiceManagerCore();
+        }
+
+        /// <summary>
+        /// Adds the minimum essential SignalR Service Manager services to the specified services collection.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        internal static IServiceCollection AddSignalRServiceManagerCore(this IServiceCollection services)
```

Comment:
is `private` enough?

Reply:
ServiceManagerBuilder calls this method.

Reply:
Now this method is merged into AddSignalRServiceManager().